File: backend/src/main/java/com/project/backend/domain/notification/dto/NotificationDTO.java
package com.project.backend.domain.notification.dto;


import com.project.backend.domain.notification.entity.Notification;
import com.project.backend.domain.notification.entity.NotificationType;
import com.project.backend.domain.notification.service.NotificationService;
import lombok.*;

import java.time.LocalDateTime;


@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class NotificationDTO {
    Long id;

    Long producerMemberId;

    Long consumerMemberId;

    Long reviewId;

    Long reviewCommentId;

    boolean isCheck;

    String content;

    NotificationType notificationType;

    LocalDateTime create_At;

    public NotificationDTO(Notification notification) {
        this.id = notification.getId();
        this.producerMemberId = notification.getProducerMemberId();
        this.consumerMemberId = notification.getConsumerMemberId();
        this.reviewId = notification.getReviewId();
        this.reviewCommentId = notification.getReviewCommentId();
        this.isCheck = notification.isCheck();
        this.content = notification.getContent();
        this.notificationType = notification.getNotificationType();
        this.create_At = notification.getCreatedAt();
    }


}

File: backend/src/main/java/com/project/backend/domain/notification/dto/NotificationDTO.kt
package com.project.backend.domain.notification.dto

import com.project.backend.domain.notification.entity.Notification
import com.project.backend.domain.notification.entity.NotificationType
import java.time.LocalDateTime

data class NotificationDTO(
    val id: Long?,
    val producerMemberId: Long?,
    val consumerMemberId: Long?,
    val reviewId: Long?,
    val reviewCommentId: Long?,
    val isCheck: Boolean,
    val content: String,
    val notificationType: NotificationType,
    val createdAt: LocalDateTime?
) {
    companion object {
        fun from(notification: Notification): NotificationDTO {
            return NotificationDTO(
                id = notification.id,
                producerMemberId = notification.producerMemberId,
                consumerMemberId = notification.consumerMemberId,
                reviewId = notification.reviewId,
                reviewCommentId = notification.reviewCommentId,
                isCheck = notification.isCheck,
                content = notification.content,
                notificationType = notification.notificationType,
                createdAt = notification.getCreatedAtSafe()
            )
        }
    }
}
